### PR TITLE
Remove exact duplicates from the compatibility suites

### PR DESCRIPTION
**Pull request by kpet**
Contributes to #494.

Signed-off-by: Kévin Petit <kpet@free.fr>

### CODE REVIEW

**Comment by Jeremy-Kemp** [comment]
Are we restricting this to be _exact_ only? If not, tests like api/test_binary are very similar as well and could be removed.

**Comment by kpet** [comment]
No. I just started with exact duplicates because they're the easiest to spot. I'll be going from most obvious to least obvious and, yes, I agree that many other tests should be removed.

File: test_conformance/compatibility/test_conformance/basic/CMakeLists.txt [hunk 1]
```diff
@@ -39,16 +37,11 @@ set(${MODULE_NAME}_SOURCES
...
     test_async_copy.cpp
     test_sizeof.c
-    test_vector_creation.cpp
     test_vec_type_hint.c
-    test_numeric_constants.cpp
     test_constant_source.cpp
     test_bufferreadwriterect.c
     test_async_strided_copy.cpp
-    test_preprocessors.cpp
     test_kernel_memory_alignment.cpp
```
Comment (Jeremy-Kemp) [inline test_conformance/compatibility/test_conformance/basic/CMakeLists.txt]: This one only has white space diffs. We could remove that too?

Comment (kpet) [reply test_conformance/compatibility/test_conformance/basic/CMakeLists.txt]: I will remove all of these in the next change. I'm trying to slice this into small-ish, (kind of) reviewable chunks.

**Comment by kpet** [comment]
Thanks for the review!

**Pull request merged**